src/main/java/ch/njol/skript/events/EvtExplode.java

```diff
@@ -0,0 +1,177 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.effects.EffExplosion;
+import ch.njol.skript.entity.EntityType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Color;
+import ch.njol.skript.util.ColorRGB;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.FireworkEffect;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityExplodeEvent;
+import org.bukkit.event.entity.FireworkExplodeEvent;
+import org.bukkit.inventory.meta.FireworkMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class EvtExplode extends SkriptEvent {
```

Comment:
I'm personally against this event containing firework explode, I would rather see the existing explode event be combined with BlockExplodeEvent (respawn anchors, beds) and EntityExplodeEvent (creepers, tnt)

firework exploding handles a completely different set of things compare to explode events

Reply:
1. Fireworks are entities, so they belong here.
2. Because firework is an entity, not merging the events will make the parser pick the new explode event without colour support. The only way around this is to add an awkward word to stop separate these two.

Reply:
1. okay, but I do not agree that it belongs here, when I hear explosion I think boom not "pop", so when I do anything with it I'm expecting it as a player to be related to tnt/beds in nether not some run away firework
2. I understand the parser issue, and I do agree it's probably something that needs some alternative but the way this is set up you've basically said screw block explode event which makes this event more annoying to use, as now the only way to correctly implement it is two events. it would make much more sense to have syntax for `on [block|entity] explo(de|)sion)` than firework being here.
3. there's only 6 entities at best in this game that cause an entity explode event so the support of `%entitydata%` doesn't make any sense to me, if it was needed an event-value for entity would work just as well.